plugin/trino-ranger/src/main/java/io/trino/plugin/ranger/RangerSystemAccessControl.java

```diff
@@ -0,0 +1,1252 @@
+    {
+        if (!hasPermission(createResource(queryId), identity, queryId, EXECUTE, "ExecuteQuery")) {
+            denyExecuteQuery();
+        }
+    }
+
+    @Override
+    public void checkCanViewQueryOwnedBy(Identity identity, Identity queryOwner)
+    {
+        if (!hasPermission(createUserResource(queryOwner.getUser()), identity, null, IMPERSONATE, "ViewQueryOwnedBy")) {
+            denyImpersonateUser(identity.getUser(), queryOwner.getUser());
+        }
+    }
+
+    @Override
+    public Collection<Identity> filterViewQueryOwnedBy(Identity identity, Collection<Identity> queryOwners)
+    {
+        Set<Identity> toExclude = new HashSet<>();
+
+        for (Identity queryOwner : queryOwners) {
+            if (!hasPermissionForFilter(createUserResource(queryOwner.getUser()), identity, null, IMPERSONATE, "filterViewQueryOwnedBy")) {
+                LOG.debug("filterViewQueryOwnedBy(user=%s): skipping queries owned by %s", identity, queryOwner);
+
+                toExclude.add(queryOwner);
+            }
+        }
+
+        return toExclude.isEmpty() ? queryOwners : queryOwners.stream().filter(((Predicate<? super Identity>) toExclude::contains).negate()).collect(Collectors.toList());
+    }
+
+    @Override
+    public void checkCanKillQueryOwnedBy(Identity identity, Identity queryOwner)
+    {
+        if (!hasPermission(createUserResource(queryOwner.getUser()), identity, null, IMPERSONATE, "KillQueryOwnedBy")) {
+            denyImpersonateUser(identity.getUser(), queryOwner.getUser());
+        }
+    }
+
+    @Override
+    public void checkCanReadSystemInformation(Identity identity)
+    {
+        if (!hasPermission(createSystemInformation(), identity, null, READ_SYSINFO, "ReadSystemInformation")) {
+            denyReadSystemInformationAccess();
+        }
+    }
+
+    @Override
+    public void checkCanWriteSystemInformation(Identity identity)
+    {
+        if (!hasPermission(createSystemInformation(), identity, null, WRITE_SYSINFO, "WriteSystemInformation")) {
+            denyWriteSystemInformationAccess();
+        }
+    }
+
+    @Deprecated
+    @Override
+    public void checkCanSetSystemSessionProperty(Identity identity, String propertyName)
+    {
+        if (!hasPermission(createSystemPropertyResource(propertyName), identity, null, ALTER, "SetSystemSessionProperty")) {
+            denySetSystemSessionProperty(propertyName);
+        }
+    }
+
+    @Override
+    public void checkCanSetSystemSessionProperty(Identity identity, QueryId queryId, String propertyName)
+    {
+        if (!hasPermission(createSystemPropertyResource(propertyName), identity, queryId, ALTER, "SetSystemSessionProperty")) {
+            denySetSystemSessionProperty(propertyName);
+        }
+    }
+
+    @Override
+    public boolean canAccessCatalog(SystemSecurityContext context, String catalogName)
+    {
+        return hasPermission(createResource(catalogName), context, USE, "AccessCatalog");
+    }
```

Comment:
Agreed. I will update following methods as well to check for "any access", instead of permission for "show":
- checkCanShowSchemas()
- checkCanShowTables() 
- checkCanShowColumns()
- checkCanShowFunctions()